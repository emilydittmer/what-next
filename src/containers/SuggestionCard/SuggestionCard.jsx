import React, { Component } from "react";
import "./SuggestionCard.scss";
import { connect } from "react-redux";

class SuggestionCard extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { id, name, backgroundImg, voteAverage, overview, date } = this.props;
    return (
      <article
        className="image-container"
        style={{
          background: `url(${backgroundImg})`
        }}
      >
        <div
          className="content"
          style={{
            background: "rgba(0,0,0,.7)"
          }}
        >
          <div className="top-line">
            <h3 className="name">{name}</h3>
            <p className="average">{voteAverage}/10</p>
          </div>
          <p className="aired">First Aired: {date}</p>
          <p className="overview">{overview}</p>
        </div>
      </article>
    );
  }
}

const mapStateToProps = state => ({
  suggestions: state.suggestions
});

export default connect(mapStateToProps)(SuggestionCard);
