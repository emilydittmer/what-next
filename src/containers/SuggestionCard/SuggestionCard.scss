.suggestion-card {
  width: 500px;
  position: relative; 
  margin: 20px;
}
 
h3 { 
  position: absolute; 
  top: 200px; 
  left: 0; 
  width: 100%; 
 }
