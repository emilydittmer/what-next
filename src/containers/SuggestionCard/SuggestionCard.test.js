import React from 'react';
import { shallow } from 'enzyme';
import {
  grabWatchlist,
  deleteFromWatchlist,
  addToWatchlist
} from "../../actions";
import { SuggestionCard, mapStateToProps, mapDispatchToProps } from './SuggestionCard';

describe('SuggestionCard', () => {
  let wrapper, instance, mockFunction;
  let props = {
    suggestions: [{
      id: 1,
      name: 'The Office'
    }],
    watchlist: []
  }
  let mockEvent = {
    preventDefault: jest.fn()
  }
  let mockHandleWatchList = jest.fn()
  beforeEach(() => {
    mockFunction = jest.fn()
    wrapper = shallow(<SuggestionCard {...props.suggestions} {...props.watchlist} />);
    instance = wrapper.instance()
  })

  it('should match the snapshot', () => {
    mockHandleWatchList(mockEvent)
    expect(wrapper).toMatchSnapshot()
  })
})