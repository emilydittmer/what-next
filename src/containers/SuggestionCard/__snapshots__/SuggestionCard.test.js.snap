// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionCard should match the snapshot 1`] = `undefined`;
